server/devpi_server/model.py

```diff
@@ -1511,7 +1517,11 @@ def create_linked_entry(self, rel, basename, content_or_file, last_modified=None
             assert overwrite is None
             overwrite = sum(x.get('count', 0)
                             for x in link.get_logs() if x.get('what') == 'overwrite')
-        self.remove_links(rel=rel, basename=basename)
+        if not isinstance(content_or_file, ELink):
```

Comment:
If you create a new link and pass in a file as described in another comment, then this issue should go away.

The link removal is there to remove an overwritten file with same version if I'm not mistaken. I still need to test that.